stdlib/json.gr

```diff
@@ -0,0 +1,2103 @@
+        codePoint != 0x0022 &&
+        codePoint != 0x005C &&
+        codePoint < 128
+      ) {
+        Buffer.addCharFromCodePoint(codePoint, buffer)
+      } else {
+        emitEscapedCodePoint(codePoint, buffer)
+      }
+    }
+  } else if (format.escapeAllControlPoints && !format.escapeNonASCII) {
+    // If desired, in addition to the required 0..31 control points,
+    // also escape unicode control point group C1 (128-159).
+    // There could be more control points or otherwise escape worthy
+    // codepoints, but covering that would be overkill.
+    (codePoint: Number) => {
+      if (
+        codePoint > 31 &&
+        codePoint != 0x0022 &&
+        codePoint != 0x005C &&
+        codePoint < 127 ||
+        codePoint > 159
+      ) {
+        Buffer.addCharFromCodePoint(codePoint, buffer)
+      } else {
+        emitEscapedCodePoint(codePoint, buffer)
+      }
+    }
+  } else {
+    // And this is just the combination of both flags, which means
+    // doing almost the same as for the case above for
+    // escapeNonASCII=true, but also escape the ASCII control codepoint
+    // 127 (Delete).
+    (codePoint: Number) => {
+      if (
+        codePoint > 31 &&
+        codePoint != 0x0022 &&
+        codePoint != 0x005C &&
+        codePoint < 127
+      ) {
+        // fast path for chars that never need any escaping
+        Buffer.addCharFromCodePoint(codePoint, buffer)
+      } else {
+        emitEscapedCodePoint(codePoint, buffer)
+      }
+    }
+  }
+
+  let emitEscapedQuotedString = if (!format.escapeHTMLUnsafeSequences) {
+    (s: String) => {
+      Buffer.addChar('"', buffer)
+
+      // Note that it's important for performance that the closure passed to forEachCodePoint
+      // is not allocated inline here, but just once when creating the writer.
+
+      String.forEachCodePoint(emitCodePoint, s)
+
+      Buffer.addChar('"', buffer)
+    }
+  } else {
+    // Special handling for the escapeHTMLUnsafeSequences flag.
+    // Escaping a sequence requires keeping track of previous characters,
+    // which is difficult and suboptimal when using a function to iterate
+    // the input string. So we don't want to pay the price in other cases.
+    // This cannot be done just in the emitCodePoint function.
+    // It could be possible to implement more optimally, but would
+    // complicate things even more than this.
+    (s: String) => {
+      Buffer.addChar('"', buffer)
+
+      let mut prevCodePoint = 0
+
+      String.forEachCodePoint(codePoint => {
+        if (codePoint == 47) {
+          if (prevCodePoint == 60) {
+            Buffer.addChar('\\', buffer)
+            Buffer.addChar('/', buffer)
+          } else {
+            // otherwise just emit the slash as-is
+            Buffer.addChar('/', buffer)
+          }
+        } else {
+          emitCodePoint(codePoint)
+        }
+
+        prevCodePoint = codePoint
+      }, s)
+
+      Buffer.addChar('"', buffer)
+    }
+  }
+
+  let implHelper = {
```

Comment:
Better name for this now?

Reply:
thoughts on `printConfig`?

Reply:
Let's leave it. A lot of the naming in this module is sub-par after all the refactoring